src/vis/plots/forecast_map.py

```diff
@@ -0,0 +1,361 @@
+from django.utils.translation import gettext_lazy as _
+from django.http import HttpRequest
+
+from main.utils import UF_CODES
+from vis.utils import geo_obj_to_dataframe, obj_to_dataframe
+from vis.models import ResultsProbForecast, GeoMacroSaude, State, Macroregion
+
+code_to_state = {v: k for k, v in UF_CODES.items()}
+
+
+def macro_forecast_map_table(
+    date: datetime.date,
+    request: HttpRequest,
+    macroregion: Optional[Literal[1, 2, 3, 4, 5]] = None,
+    # fmt: off
+    uf: Optional[
+        Literal[
+            "AC", "AL", "AP", "AM", "BA", "CE", "ES", "GO", "MA",
+            "MT", "MS", "MG", "PA", "PB", "PR", "PE", "PI", "RJ",
+            "RN", "RS", "RO", "RR", "SC", "SP", "SE", "TO", "DF",
+        ]
+    ] = None,
+    # fmt: on
+    geocodes: Optional[list[int]] = None,
+    title_left: str = "Limiar superior de Incidência na semana",
+    title_right: str = "Previsão probabilística na semana",
+    fontsize: int = 16,
+    width: int = 450,
+    height: int = 350,
+    table_width: int = int(450 * 0.25),
+):
+    if not isinstance(date, datetime.date):
+        raise ValueError("date must be a datetime.date object")
+
+    macros_saude = GeoMacroSaude.objects.all()
+
+    if geocodes:
+        macros_saude = GeoMacroSaude.objects.filter(geocode__in=geocodes)
+        if not macros_saude:
+            raise ValueError(
+                "Incorrect MacroSaude geocode(s). Example: [1101, 1102]"
+            )
+
+    if uf:
+        try:
+            macros_saude = GeoMacroSaude.objects.filter(
+                state=State.objects.get(uf=uf)
+            )
+        except State.DoesNotExist:
+            raise ValueError("Incorrect UF. Example: 'SP'")
+
+    if macroregion:
+        macros_saude = GeoMacroSaude.objects.filter(
+            state__in=(
+                State.objects.filter(
+                    macroregion=(
+                        Macroregion.objects.get(geocode=str(macroregion))
+                    )
+                )
+            )
+        )
+        if not macros_saude:
+            raise ValueError("Incorrect macroregion selected. Options: 1 to 5")
+
+    results = set()
+    for macro in macros_saude:
+        results.add(
+            get_object_or_404(ResultsProbForecast, date=date, geocode=macro)
+        )
+
+    df = pd.concat([obj_to_dataframe(o) for o in results]).reset_index(
+        drop=True
+    )
+
+    df_macro = gpd.GeoDataFrame(
+        pd.concat(
+            [geo_obj_to_dataframe(o) for o in macros_saude],
+        )
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+
+    df["date"] = df["date"].apply(str)
+
+    df.prob_low = -df.prob_low
+
+    df["prob_color"] = df.apply(
+        lambda x: x.prob_low
+        if abs(x.prob_low) > abs(x.prob_high)
+        else x.prob_high,
+        axis=1,
+    )
+
+    df["prob_color"] = df.prob_color.apply(lambda x: 0 if abs(x) < 50 else x)
+
+    df = df.drop(columns="id")
```

Comment:
You can do this drop "inplace" avoiding having to copy the dataframe onto itself